gpkit/geometric_program.py

```diff
@@ -146,9 +146,10 @@ def solve(self, solver=None, verbosity=1, *args, **kwargs):
 
         result = {}
         # confirm lengths before calling zip
-        assert len(self.varlocs) == len(solver_out['primal'])
+        primal = solver_out['primal']
+        assert len(self.varlocs) == len(primal)
         result["variables"] = dict(zip(self.varlocs,
-                                       np.exp(solver_out['primal']).ravel()))
```

Comment:
I think ravel was working here before because `np.exp` was called, thus always turning it into an np array.